util/internal/integrationtest/common/src/main/java/org/hibernate/search/util/impl/integrationtest/common/extension/MappingSetupHelper.java

```diff
@@ -89,44 +48,48 @@ public String toString() {
 
 	public abstract MappingAssertionHelper<? super R> assertions();
 
+	protected abstract SV defaultSetupVariant();
+
 	public C start() {
-		C setupContext = createSetupContext();
+		return start( defaultSetupVariant() );
+	}
+
+	public C start(SV setupVariant) {
+		C setupContext = createSetupContext( setupVariant );
 		return backendSetupStrategy.start( setupContext, configurationProvider, setupContext.backendMappingHandlePromise );
 	}
 
 	@Override
 	public void afterAll(ExtensionContext context) throws Exception {
-		delegate.afterAll( context );
+		configurationProvider.afterAll( context );
+		if ( callOncePerClass ) {
+			cleanUp();
+		}
```

Comment:
I don't understand how this is supposed to work, but anyway: considering there may be multiple session factories created, potentially each at a different time (beforeall, beforeeach, test, ...), wouldn't it make more sense to attach a "scope" to every created resource, and decide whether to clean it up based on that?

E.g. you could have two `toClose` lists, `toCloseAfterTest` and `toCloseAfterAll`. In `setup` you'd put the closeable in the relevant list based on the current value of `callOncePerClass` (`currentScope` would be a better name BTW). And then you execute the corresponding cleanup in the corresponding callbacks (afterEach/afterAll).

This might be related to the problem I noticed in `SearchQueryEntityLoadingGraphIT`, BTW.